webapp/babel.config.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+const config = {
+    presets: [
+        ['@babel/preset-env', {
+            targets: {
+                chrome: 66,
+                firefox: 60,
+                edge: 42,
+                safari: 12,
+            },
+            modules: false,
+            corejs: 3,
+            debug: false,
+            useBuiltIns: 'usage',
+            shippedProposals: true,
+        }],
+        ['@babel/preset-react', {
+            useBuiltIns: true,
+        }],
+    ],
+    plugins: [
+        '@babel/plugin-proposal-class-properties',
+        '@babel/plugin-syntax-dynamic-import',
+        '@babel/proposal-object-rest-spread',
+        '@babel/plugin-proposal-optional-chaining',
+    ],
```

Comment:
Should the template file from https://github.com/mattermost/mattermost-plugin-starter-template/blob/master/webapp/babel.config.js be used here?

Reply:
It's almost the same. I've just removed typescript features as they're not used in this plugin.

Reply:
@kosgrz Do you see value in introducing Typescript to the plugin? The config we use allows for a gradual migration. For example, you can turn any file in the project to Typescript and it will compile properly with Javascript files importing it or vise versa.